kernel/kernel-api/src/test/scala/io/delta/kernel/internal/util/ColumnMappingSuite.scala

```diff
@@ -193,6 +263,246 @@ class ColumnMappingSuite extends AnyFunSuite {
+          Maps.newHashMap(TableConfig.ICEBERG_COMPAT_V2_ENABLED.getKey, "true")),
+        ID,
+        true /** isNewTable */)
+
+    assertColumnMapping(metadata.getSchema.get("a"), 1L)
+    assertColumnMapping(metadata.getSchema.get("b"), 2L)
+    val innerStruct = metadata.getSchema.get("b").getDataType.asInstanceOf[StructType]
+    assertColumnMapping(innerStruct.get("d"), 3L)
+    assertColumnMapping(innerStruct.get("e"), 4L)
+    assertColumnMapping(metadata.getSchema.get("c"), 5L)
+
+    assertThat(metadata.getConfiguration)
+      .containsEntry(ColumnMapping.COLUMN_MAPPING_MAX_COLUMN_ID_KEY, "5")
+  }
+
+  test("assign id and physical name to new table with array type and enableIcebergCompatV2=true") {
+    val schema: StructType =
+      new StructType()
+        .add("a", StringType.STRING)
+        .add("b", new ArrayType(IntegerType.INTEGER, false))
+        .add("c", IntegerType.INTEGER)
+
+    val metadata: Metadata =
+      ColumnMapping.updateColumnMappingMetadata(
+        createMetadata(schema).withNewConfiguration(
+          Maps.newHashMap(TableConfig.ICEBERG_COMPAT_V2_ENABLED.getKey, "true")),
+        ID,
+        true /** isNewTable */)
+
+    assertColumnMapping(metadata.getSchema.get("a"), 1L)
+    assertColumnMapping(metadata.getSchema.get("b"), 2L)
+    assertColumnMapping(metadata.getSchema.get("c"), 3L)
+
+    // verify nested ids
+    assertThat(metadata.getSchema.get("b").getMetadata.getEntries
+      .get(COLUMN_MAPPING_NESTED_IDS_KEY).asInstanceOf[FieldMetadata].getEntries)
+      .hasSize(1)
+      .anySatisfy((k: AnyRef, v: AnyRef) => {
+        assertThat(k).asString.startsWith("col-")
+        assertThat(k).asString.endsWith(".element")
+        assertThat(v).isEqualTo(4L)
+      })
+
+    assertThat(metadata.getConfiguration)
+      .containsEntry(ColumnMapping.COLUMN_MAPPING_MAX_COLUMN_ID_KEY, "4")
+  }
+
+  test("assign id and physical name to existing table with array type " +
+    "and enableIcebergCompatV2=true") {
+    val schema: StructType =
+      new StructType()
+        .add("a", StringType.STRING)
+        .add("b", new ArrayType(IntegerType.INTEGER, false))
+        .add("c", IntegerType.INTEGER)
+
+    val metadata: Metadata =
+      ColumnMapping.updateColumnMappingMetadata(
+        createMetadata(schema).withNewConfiguration(
+          Maps.newHashMap(TableConfig.ICEBERG_COMPAT_V2_ENABLED.getKey, "true")),
+        ID,
+        false /** isNewTable */)
+
+    assertColumnMapping(metadata.getSchema.get("a"), 1L, "a")
+    assertColumnMapping(metadata.getSchema.get("b"), 2L, "b")
+    assertColumnMapping(metadata.getSchema.get("c"), 3L, "c")
+
+    // verify nested ids
+    assertThat(metadata.getSchema.get("b").getMetadata.getEntries
+      .get(COLUMN_MAPPING_NESTED_IDS_KEY).asInstanceOf[FieldMetadata].getEntries)
+      .hasSize(1)
+      .anySatisfy((k: AnyRef, v: AnyRef) => {
+        assertThat(k).asString.isEqualTo("b.element")
+        assertThat(v).isEqualTo(4L)
+      })
```

Comment:
utility methods which takes a verification method as arg? seems like this is repeated.

Reply:
actually we don't need to use `anySatisfy` here and can just use `containsEntry("b.element", 4L)` directly. The only other place(s) we use `anySatisfy` is where the key contains a generated UUID but I don't think it's worth introducing a util method just for this check